pulp_container/tests/functional/api/test_token_authentication.py

```diff
@@ -121,6 +121,10 @@ def test_pull_image_with_real_container_client(self):
 
         image_url = urljoin(self.cfg.get_base_url(), self.distribution.base_path)
         image_with_tag = f"{image_url}:manifest_a"
+
+        registry_name = urlparse(self.cfg.get_base_url()).netloc
+        admin_user, admin_password = self.cfg.pulp_auth
+        registry.login("-u", admin_user, "-p", admin_password, registry_name)
```

Comment:
I don't know how this test worked before. 
I believe my changes didn't affect it. If you have any thoughts, please do share.

My wild guess is that it ran in some order that the user was logged into registry from previous tests.
Maybe because in my tests I explicitly logout, it started failing. :shrug:

Reply:
before rbac anynoumos user could pull, now that we have rbac + we logout after every operation it is very possible that your wild guess is not that wild ;)